posixlib/src/main/scala/scala/scalanative/runtime/SocketHelpers.scala

```diff
@@ -19,17 +19,14 @@ object SocketHelpers {
 
   def isReachableByEcho(ip: String, timeout: Int, port: Int): Boolean = {
     Zone { implicit z =>
-      val cIP   = toCString(ip)
-      var hints = alloc[addrinfo]
-      var ret   = alloc[Ptr[addrinfo]]
+      val cIP = toCString(ip)
 
-      libc.memset(hints.rawptr, 0, sizeof[addrinfo])
+      var hints = alloc[addrinfo] // alloc documented to clear retured memory
```

Comment:
Both `hints` and `ret` are only assigned once, they can use `val` instead.

Reply:
I believe that you are correct for `hints`.  I believe that `ret` needs to be a `var`, since
it gets changed/written_into by the `getaddrinfo()` call (**res). 
```
   int getaddrinfo(const char *node, const char *service,
                       const struct addrinfo *hints,
                       struct addrinfo **res);

```
The current scala declaratiion is: `var ret = alloc[Ptr[addrinfo]]`
it should probably be `var ret = alloc[Ptr[Ptr[addrinfo]]]`
If you concur, I can correct that in several places, possibly
also in PlainSocketImpl.scala.